src/Lookup/UnresolvedEntityRedirectException.php

```diff
@@ -23,9 +24,15 @@ class UnresolvedEntityRedirectException extends EntityLookupException {
 	/**
 	 * @param EntityId $entityId
 	 * @param EntityId $redirectTargetId The ID of the target Entity of the redirect
+	 * @param string|null $message
+	 * @param Exception|null $previous
 	 */
-	public function __construct( EntityId $entityId, EntityId $redirectTargetId ) {
-		parent::__construct( $entityId, 'Unresolved redirect to ' . $redirectTargetId->getSerialization() );
+	public function __construct( EntityId $entityId, EntityId $redirectTargetId, $message = null, Exception $previous = null ) {
```

Comment:
So when a custom `$message` is provided, the `$redirectTargetId` is ignored. So `$message` does have the higher precedence, but comes later in the list of parameters and is optional. `$redirectTargetId` on the other hand is mandatory, even if a custom `$message` is provided and `$redirectTargetId` is ignored. Still a -1 for this.